pkg/metrics/client_go_adapter.go

```diff
@@ -30,6 +32,26 @@ import (
 var (
 	// client metrics.
 
+	requestLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "rest_client_request_duration_seconds",
+			Help: "Request latency in seconds. Broken down by verb and URL.",
+			Buckets: []float64{0.005, 0.01, 0.025, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0,
+				1.25, 1.5, 1.75, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5, 6, 7, 8, 9, 10, 15, 20, 25, 30, 40, 50, 60},
+		},
+		[]string{"verb", "host", "path"},
+	)
+
+	rateLimiterLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "rest_client_rate_limiter_duration_seconds",
+			Help: "Client side rate limiter latency in seconds. Broken down by verb and URL.",
+			Buckets: []float64{0.005, 0.01, 0.025, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0,
```

Comment:
These have a ton of buckets and dimensions that include the path which means things like object names will be in there, this will result in a huge cardinality, we can not include something like this by default

Reply:
My primary requirement is to register some custom metrics for client-go/tools/metrics. However, when using controller-runtime, I am unable to customize the registration of client-go metrics because registerClientMetrics() has a higher precedence. Therefore, I need to add some additional metrics. Perhaps we could reduce the number of buckets and some tags metrics in this case?